### PR TITLE
fix #2188. Reset the offset to 0 on Table refresh

### LINKED ISSUE #2188: Reset the offset to 0 on Table refresh? ('refresh')
How can i reset the offset to 0 on Table refresh? ('refresh')

### PR DESCRIPTION
**Pull request by tangjn**
Use refresh example:
$('#table').bootstrapTable('refresh', {resetOffset: true});

### CODE REVIEW

File: src/bootstrap-table.js [hunk 1]
```diff
@@ -1926,10 +1928,16 @@
             };
 
             if (this.options.pagination) {
-                params.offset = this.options.pageSize === this.options.formatAllRows() ?
-                    0 : this.options.pageSize * (this.options.pageNumber - 1);
                 params.limit = this.options.pageSize === this.options.formatAllRows() ?
                     this.options.totalRows : this.options.pageSize;
+                // fix #2188. Reset the offset to 0 on Table refresh
+                if(this.options.resetOffset === true){
```
Comment (wenzhixin) [inline src/bootstrap-table.js]: `if (this.options.resetOffset) {`

File: src/bootstrap-table.js [hunk 2]
```diff
@@ -1926,10 +1928,16 @@
...
+                    params.offset = 0;
+                    this.options.resetOffset = false;
+                }else{
```
Comment (wenzhixin) [inline src/bootstrap-table.js]: `} else {`

File: src/bootstrap-table.js [hunk 3]
```diff
@@ -2690,6 +2698,11 @@
     };
 
     BootstrapTable.prototype.refresh = function (params) {
+        // fix #2188. Reset the offset to 0 on Table refresh
+        if(params.resetOffset === true){
```
Comment (wenzhixin) [inline src/bootstrap-table.js]: `if (params && params.resetOffset) {`
`params` maybe `undefined`, will cause error:
```
Uncaught TypeError: Cannot read property 'resetOffset' of undefined
```

**Comment by wenzhixin** [comment]
Also need to update https://github.com/tangjn/bootstrap-table/blob/f78c1f6fed5b7ef0518fdd42194e5f3c486515e7/src/bootstrap-table.js#L1914.

Please help to update the documentation: https://github.com/wenzhixin/bootstrap-table/blob/develop/docs/_i18n/en/documentation/table-options.md

**Comment by tangjn** [comment]
@wenzhixin  I don't understand why need to update https://github.com/tangjn/bootstrap-table/blob/f78c1f6fed5b7ef0518fdd42194e5f3c486515e7/src/bootstrap-table.js#L1914?

**Comment by wenzhixin** [comment]
When `queryParamsType !== 'limit'`

**Comment by tangjn** [comment]
Sorry, can you give more detail information?

**Comment by wenzhixin** [comment]
Fixed in https://github.com/wenzhixin/bootstrap-table/commit/166b613e211d9a19cc759053672fc05867df586f, you can use:
```js
$table.bootstrapTable({pageNumber: 1});
```
to solve your problem, thanks for your contribution!

**Pull request closed without merging**